File: exemples/sendRecvEx1.c
#include <stdio.h>
#include <mpi.h>

int main (int argc, char * argv[]) {
    int rank, size, buffer = -1, tag = 10;
    MPI_Status status;

    MPI_Init( &argc, &argv );
    MPI_Comm_rank( MPI_COMM_WORLD, &rank );
    MPI_Comm_size( MPI_COMM_WORLD, &size );

    if (size >= 2 && rank == 0) {
        buffer = 33;
        MPI_Send( &buffer, 1, MPI_INT, 1, tag, MPI_COMM_WORLD );
    }
    if (size >= 2 && rank == 1) {
        MPI_Recv( &buffer, 1, MPI_INT, 0, tag, MPI_COMM_WORLD, &status );
        printf("Rank %d\tbuffer= %d\n", rank, buffer);
        if (buffer != 33) printf("fail\n");
    }

    MPI_Finalize();
    return 0;
}

File: demos/sendRecvEx1.f90
PROGRAM sendRecvEx1
use mpi
IMPLICIT NONE

INTEGER :: rank, buffer, error, status( MPI_status_size )

    CALL MPI_init( error )
    CALL MPI_Comm_rank( MPI_COMM_WORLD, rank, error )

    IF( rank == 0 ) THEN
        buffer = 33
        CALL MPI_Send( buffer, 1, MPI_INTEGER, 1, 10, &
                       MPI_COMM_WORLD, error )
    END IF
    IF( rank == 1 ) THEN
        CALL MPI_Recv( buffer, 1, MPI_INTEGER, 0, 10, &
            MPI_COMM_WORLD, status, error )
        PRINT*, 'Rank ', rank, ' buffer=', buffer
        IF( buffer /= 33 ) Print*, 'fail'
    END IF

    CALL MPI_Finalize( error )
END PROGRAM sendRecvEx1
